packages/wrangler/src/versions/api.ts

```diff
@@ -104,28 +108,62 @@ export async function createDeployment(
 	accountId: string,
 	workerName: string,
 	versionTraffic: Map<VersionId, Percentage>,
-	message: string | undefined
+	message: string | undefined,
+	force?: boolean
 ) {
-	const res = await fetchResult(
-		`/accounts/${accountId}/workers/scripts/${workerName}/deployments`,
-		{
-			method: "POST",
-			headers: { "Content-Type": "application/json" },
-			body: JSON.stringify({
-				strategy: "percentage",
-				versions: Array.from(versionTraffic).map(
-					([version_id, percentage]) => ({ version_id, percentage })
-				),
-				annotations: {
-					"workers/message": message,
-				},
-			}),
+	try {
+		return await fetchResult(
+			`/accounts/${accountId}/workers/scripts/${workerName}/deployments${force ? "?force=true" : ""}`,
+			{
+				method: "POST",
+				headers: { "Content-Type": "application/json" },
+				body: JSON.stringify({
+					strategy: "percentage",
+					versions: Array.from(versionTraffic).map(
+						([version_id, percentage]) => ({ version_id, percentage })
+					),
+					annotations: {
+						"workers/message": message,
+					},
+				}),
+			}
+		);
+	} catch (e) {
+		if (
+			e instanceof APIError &&
+			e.code === CANNOT_ROLLBACK_WITH_MODIFIED_SECERT_CODE
+		) {
+			// This is not great but is the best way I could think to handle for now
+			const errorMsg = e.notes[0].text.replace(
+				` [code: ${CANNOT_ROLLBACK_WITH_MODIFIED_SECERT_CODE}]`,
+				""
+			);
+			const targetString = "The following secrets have changed:";
+			const changedSecrets = errorMsg
```

Comment:
are we guaranteed the message will have this format or that `e.notes[0].text` will always be returned?

Reply:
Yes, if this error code is returned. This error will always be this way.

This was the best way I could think to pass the secret list down to Wrangler while also giving a helpful error to users so it's a bit jank but does work

Reply:
yeah not sure it is :/
to me that's less clear